yt_dlp/extractor/rokfin.py

```diff
@@ -100,6 +100,10 @@ def _real_extract(self, url):
                        else 'not_live')
 
         video_url = traverse_obj(metadata, 'url', ('content', 'contentUrl'), expected_type=url_or_none)
+        timeline_url = traverse_obj(metadata, 'timelineUrl', ('content', 'timelineUrl'), expected_type=url_or_none)
+        if (not video_url or video_url == 'fake.m3u8') and timeline_url:
+            video_url = re.sub('https://image.v.rokfin.com/([^/]+)/storyboard.vtt$', 'https://stream.v.rokfin.com/\\1.m3u8', timeline_url)
```

Comment:
If they change the url format, instead of failing, this would just put `video_url = timeline_url` which we don't want. Maybe instead of `re.sub`, `_search_regex` the slug out first?

Like this, for e.g.
```suggestion
        if video_url in (None, 'fake.m3u8'):
            video_url = format_field(self._search_regex(
                r'https?://[^/]+/([^/]+)/storyboard.vtt',
                traverse_obj(metadata, 'timelineUrl', ('content', 'timelineUrl'), expected_type=url_or_none),
                video_id, default=None), None, 'https://stream.v.rokfin.com/%s.m3u8')
```